### PR TITLE
refactor & CI with GithubActions

### PR DESCRIPTION
**Pull request by s-sajid-ali**
- Updated README with relevant changes.
- Add `-pdb` flag to updater to aid in debugging.
- Add CI via GithubActions.
- Switch to `pytest` for testing and use files from `https://github.com/openPMD/openPMD-example-datasets`. More test cases to be added later.
- Allow data fields in openPMD files to be optional.
- Use modern PEP-517 based flit build system.

### CODE REVIEW

**Comment by s-sajid-ali** [comment]
Ping @ax3l . This can now be reviewed and merged. I'll add the test for other files in a separate PR.

**Comment by s-sajid-ali** [comment]
I've added a basic CI workflow, but the repository owner needs to enable GithubActions before it can proceed. If you could enable this, I'll ensure that the CI I've added works as expected.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+on: [push]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.10", "3.11", "3.12"]
```
Comment (ax3l) [inline .github/workflows/ci.yml:11]:
```suggestion
        python-version: ["3.10", "3.11", "3.12"]
```

File: openpmd_updater/cli.py [hunk 2]
```diff
@@ -1,61 +1,63 @@
...
-def main():
-    file_name, verbose, target_version, in_place = parse_cmd(sys.argv[1:])
 
-    updater = Updater(file_name, verbose)
-    updater.update(target_version, in_place)
+def parse_args():
+    help_str = (
+        "This is the openPMD updater HDF5 files.\n"
+        + "It allows to update openPMD flavored files from openPMD standard {0} to {1}\n".format(
+            111, 222
```
Comment (ax3l) [inline openpmd_updater/cli.py:20]: What is 111 and 222? :D

File: openpmd_updater/cli.py [hunk 3]
```diff
@@ -1,61 +1,63 @@
...
+        )
+    )
+    parser = argparse.ArgumentParser(
+        description=help_str, formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+    parser.add_argument("file", help="OpenPMD-series-filename", type=str)
```
Comment (ax3l) [inline openpmd_updater/cli.py:26]:
```suggestion
    parser.add_argument("file", help="openPMD-series-filename", type=str)
```

File: openpmd_updater/cli.py (hunk 2 above)
Comment (s-sajid-ali) [reply openpmd_updater/cli.py:20]: Holdover from existing source https://github.com/openPMD/openPMD-updater/blob/1f334c99093b7175af990d57879ea24ff5b4cb01/openpmd_updater/cli.py#L17 

should it be changed to 1.1.0 and 2.0.0?

Comment (ax3l) [reply openpmd_updater/cli.py:20]: Yes, I am not sure what it prints now without running it.

Comment (s-sajid-ali) [reply openpmd_updater/cli.py]: Fixed in https://github.com/openPMD/openPMD-updater/pull/10/commits/7c527f940555a92a6f197971b44cad2e87ff67b7.

**Review by ax3l — APPROVED** [review]
Thanks a lot! :)

**Pull request merged**